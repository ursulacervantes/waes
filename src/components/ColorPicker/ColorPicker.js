import React, { useState } from 'react';

import ButtonToolbar from 'react-bootstrap/ButtonToolbar';
import Button from 'react-bootstrap/Button';

import './ColorPicker.scss';

const ColorPicker = props => {
  const [colors, setStatus] = useState([
    {
      color: 'red',
      status: ''
    },
    {
      color: 'yellow',
      status: ''
    },
    {
      color: 'green',
      status: ''
    }
  ]);

  const handleChange = color => {
    setStatus(currentColor => {
      return currentColor.map(val => {
        if (props.singlePick) {
          if (val.color !== color) {
            val.status = '';
          }
        }

        if (val.color === color) {
          val.status = val.status === 'focus' ? '' : 'focus';
        }

        return val;
      });
    });
  };

  const onMouseDown = e => {
    e.preventDefault();
  };

  return (
    <div>
      <ButtonToolbar>
        {colors.map((item, key) => (
          <Button
            type="button"
            key={item.color}
            className={`button ${item.color} ${item.status}`}
            onClick={() => handleChange(item.color)}
            onMouseDown={onMouseDown}
          ></Button>
        ))}
      </ButtonToolbar>
    </div>
  );
};

export default ColorPicker;
