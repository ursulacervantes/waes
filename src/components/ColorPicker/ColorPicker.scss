@import 'bootstrap';
@import '../../variables';

.button {
  margin-right: 5px;
  padding: 0.75rem 1.125rem !important;
}

.red {
  @include button-variant(
    $red,
    darken($red, 7.5%),
    darken($red, 10%),
    lighten($red, 5%),
    lighten($red, 10%),
    darken($red, 30%)
  );
}

.yellow {
  @include button-variant(
    $yellow,
    darken($yellow, 7.5%),
    darken($yellow, 10%),
    lighten($yellow, 5%),
    lighten($yellow, 10%),
    darken($yellow, 30%)
  );
}

.green {
  @include button-variant(
    $green,
    darken($green, 7.5%),
    darken($green, 10%),
    lighten($green, 5%),
    lighten($green, 10%),
    darken($green, 30%)
  );
}
