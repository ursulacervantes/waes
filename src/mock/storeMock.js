const color = {
  editor: [
    {
      color: 'red',
      status: ''
    },
    {
      color: 'yellow',
      status: ''
    },
    {
      color: 'green',
      status: ''
    }
  ],
  filter: [
    {
      color: 'red',
      status: ''
    },
    {
      color: 'yellow',
      status: ''
    },
    {
      color: 'green',
      status: ''
    }
  ]
};

const text = {
  red: {},
  yellow: {},
  green: {}
};

export const initialState = { color, text };
