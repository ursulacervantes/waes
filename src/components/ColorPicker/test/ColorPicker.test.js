import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import ColorPicker from '../../ColorPicker';
import renderer from 'react-test-renderer';
import { initialState } from '../../../mock/storeMock';

let state = initialState;
const mockStore = configureMockStore();
const store = mockStore(() => state);

function setColor(color) {}

describe('Color Picker', () => {
  it('select only one color', () => {
    const component = renderer.create(
      <Provider store={store}>
        <ColorPicker
          singlePick={true}
          colorId="editor"
          setColor={setColor}
        ></ColorPicker>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // // manually trigger red color button
    tree.children[0].children[0].props.onClick();
    state.color.editor = [
      {
        color: 'red',
        status: 'focus'
      },
      {
        color: 'yellow',
        status: ''
      },
      {
        color: 'green',
        status: ''
      }
    ];
    store.dispatch({ type: 'SELECT_COLOR' });

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger yellow color button
    tree.children[0].children[1].props.onClick();
    state.color.editor = [
      {
        color: 'red',
        status: ''
      },
      {
        color: 'yellow',
        status: 'focus'
      },
      {
        color: 'green',
        status: ''
      }
    ];
    store.dispatch({ type: 'SELECT_COLOR' });

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('select multiple colors', () => {
    const component = renderer.create(
      <Provider store={store}>
        <ColorPicker
          singlePick={false}
          colorId="filter"
          setColor={setColor}
        ></ColorPicker>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // // manually trigger red color button
    tree.children[0].children[0].props.onClick();
    state.color.filter = [
      {
        color: 'red',
        status: 'focus'
      },
      {
        color: 'yellow',
        status: ''
      },
      {
        color: 'green',
        status: ''
      }
    ];
    store.dispatch({ type: 'SELECT_COLOR' });

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger yellow color button
    tree.children[0].children[1].props.onClick();
    state.color.filter = [
      {
        color: 'red',
        status: 'focus'
      },
      {
        color: 'yellow',
        status: 'focus'
      },
      {
        color: 'green',
        status: ''
      }
    ];
    store.dispatch({ type: 'SELECT_COLOR' });

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
