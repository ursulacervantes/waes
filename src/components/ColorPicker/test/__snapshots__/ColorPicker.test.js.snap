// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Picker select multiple colors 1`] = `
<div>
  <div
    className="btn-toolbar"
    role="toolbar"
  >
    <button
      className="button red  btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
    <button
      className="button yellow  btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
    <button
      className="button green  btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Color Picker select multiple colors 2`] = `
<div>
  <div
    className="btn-toolbar"
    role="toolbar"
  >
    <button
      className="button red focus btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
    <button
      className="button yellow  btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
    <button
      className="button green  btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Color Picker select multiple colors 3`] = `
<div>
  <div
    className="btn-toolbar"
    role="toolbar"
  >
    <button
      className="button red focus btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
    <button
      className="button yellow focus btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
    <button
      className="button green  btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Color Picker select only one color 1`] = `
<div>
  <div
    className="btn-toolbar"
    role="toolbar"
  >
    <button
      className="button red  btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
    <button
      className="button yellow  btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
    <button
      className="button green  btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Color Picker select only one color 2`] = `
<div>
  <div
    className="btn-toolbar"
    role="toolbar"
  >
    <button
      className="button red focus btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
    <button
      className="button yellow  btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
    <button
      className="button green  btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Color Picker select only one color 3`] = `
<div>
  <div
    className="btn-toolbar"
    role="toolbar"
  >
    <button
      className="button red  btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
    <button
      className="button yellow focus btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
    <button
      className="button green  btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    />
  </div>
</div>
`;
